tensorboard/webapp/app_routing/effects/app_routing_effects_test.ts

```diff
@@ -587,190 +633,183 @@ describe('app_routing_effects', () => {
         ]);
       });
     });
-  });
-
-  describe('changeBrowserUrl$', () => {
-    let replaceStateSpy: jasmine.Spy;
 
-    beforeEach(() => {
-      effects = TestBed.inject(AppRoutingEffects);
-      effects.changeBrowserUrl$.subscribe(() => {});
-      replaceStateSpy = spyOn(location, 'replaceState');
-    });
+    describe('url changes', () => {
+      let replaceStateSpy: jasmine.Spy;
 
-    it('noops if the new route matches current URL', () => {
-      const activeRoute = buildRoute({
-        routeKind: RouteKind.EXPERIMENTS,
-        pathname: '/experiments',
-        queryParams: [],
-        navigationOptions: {
-          replaceState: false,
-        },
+      beforeEach(() => {
+        replaceStateSpy = spyOn(location, 'replaceState');
       });
-      store.overrideSelector(getActiveRoute, activeRoute);
-      store.refreshState();
-      getHashSpy.and.returnValue('');
-      getPathSpy.and.returnValue('/experiments');
-      getSearchSpy.and.returnValue([]);
 
-      action.next(
-        actions.navigated({
-          before: null,
-          after: activeRoute,
-        })
-      );
+      function navigateAndExpect(
+        navigation: Navigation,
+        expected: {pushStateUrl: null | string; replaceStateUrl: null | string}
+      ) {
+        fakeAsync(() => {
+          action.next(actions.navigationRequested(navigation));
 
-      expect(pushStateSpy).not.toHaveBeenCalled();
-      expect(replaceStateSpy).not.toHaveBeenCalled();
-    });
+          tick();
+          if (expected.pushStateUrl === null) {
+            expect(pushStateSpy).not.toHaveBeenCalled();
+          } else {
+            expect(pushStateSpy).toHaveBeenCalledWith(expected.pushStateUrl);
+          }
+
+          if (expected.replaceStateUrl === null) {
+            expect(replaceStateSpy).not.toHaveBeenCalled();
+          } else {
+            expect(replaceStateSpy).toHaveBeenCalledWith(
+              expected.replaceStateUrl
+            );
+          }
+        })();
+      }
+
+      it('noops if the new route matches current URL', () => {
+        const activeRoute = buildRoute({
+          routeKind: RouteKind.EXPERIMENTS,
+          pathname: '/experiments',
+          queryParams: [],
+          navigationOptions: {
+            replaceState: false,
+          },
+        });
+        store.overrideSelector(getActiveRoute, activeRoute);
+        store.refreshState();
+        getHashSpy.and.returnValue('');
+        getPathSpy.and.returnValue('/experiments');
+        getSearchSpy.and.returnValue([]);
 
-    it('pushes state if path and search do not match new route on navigated', () => {
-      const activeRoute = buildRoute({
-        routeKind: RouteKind.EXPERIMENTS,
-        pathname: '/experiments',
-        queryParams: [],
-        navigationOptions: {
-          replaceState: false,
-        },
+        navigateAndExpect(activeRoute, {
```

Comment:
Based on this call, the type for the first parameter to `navigateAndExpect` on L645 should be `Navigation | Route`, not `Navigation`.

Reply:
Fixed; thanks for catching it.